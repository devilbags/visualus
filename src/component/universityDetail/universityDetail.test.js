import React from 'react';
import { shallow } from 'enzyme';
import UniversityDetail from './universityDetail';

describe('UniversityDetail component', () => {
  const university = {
    id: 1,
    detail: {
      student: {
        size: {
          '2010': 1
        }
      }
    }
  };
  let wrapper;
  it('should render without crashing', () => {
    shallow(<UniversityDetail university={university} />);
  });
  it('should render no detail message when u detail is not available', () => {
    const u = {
      id: 1
    };
    wrapper = shallow(<UniversityDetail university={u} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render StudentSize when detail is availabel', () => {
    wrapper = shallow(<UniversityDetail university={university} />);
    expect(wrapper).toMatchSnapshot();
  });
});
