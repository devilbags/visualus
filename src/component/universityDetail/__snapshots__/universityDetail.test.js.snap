// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UniversityDetail component should render StudentSize when detail is availabel 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UniversityDetail
    university={
        Object {
            "detail": Object {
              "student": Object {
                "size": Object {
                  "2010": 1,
                },
              },
            },
            "id": 1,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1 />,
        <StudentSize
          detail={
                    Object {
                              "student": Object {
                                "size": Object {
                                  "2010": 1,
                                },
                              },
                            }
          }
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "detail": Object {
            "student": Object {
              "size": Object {
                "2010": 1,
              },
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1 />,
          <StudentSize
            detail={
                        Object {
                                    "student": Object {
                                      "size": Object {
                                        "2010": 1,
                                      },
                                    },
                                  }
            }
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "detail": Object {
              "student": Object {
                "size": Object {
                  "2010": 1,
                },
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`UniversityDetail component should render no detail message when u detail is not available 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UniversityDetail
    university={
        Object {
            "id": 1,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1 />,
        <h2>
          No detail yet
</h2>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "No detail yet",
        },
        "ref": null,
        "rendered": "No detail yet",
        "type": "h2",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1 />,
          <h2>
            No detail yet
</h2>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No detail yet",
          },
          "ref": null,
          "rendered": "No detail yet",
          "type": "h2",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
